src/file_helper.rs

```diff
@@ -0,0 +1,116 @@
+use serde::{Deserialize, Serialize};
+use starcoin_crypto::HashValue;
+use starcoin_types::block::Block;
+use std::cell::Cell;
+use std::fmt::Debug;
+use std::fs;
+use std::fs::File;
+use std::io::{BufReader, BufWriter, Read, Write};
+
+static mut STEPS: Cell<u32> = Cell::new(0u32);
+
+pub fn skip_step() {
+    unsafe {
+        get_and_inc();
+    }
+}
+
+pub fn init_file_path(block_hash: HashValue) -> std::io::Result<()> {
+    fs::create_dir_all(block_hash.to_hex())
+}
+
+pub fn serialize_to_file<T, K>(block_hash: HashValue, key: &K, obj: &T) -> std::io::Result<()>
+where
+    T: ?Sized + Serialize + Debug,
+    K: ?Sized + Serialize + Debug,
+{
+    let mut key_bytes = bcs_ext::to_bytes(key).unwrap();
+    unsafe {
+        append_nonce(&mut key_bytes);
+    }
+    let file_name = format!(
+        "./{}/{}",
```

Comment:
make the dir configurable.

Reply:
I think the dir should be transparent to the user, they don't care at all, so, maybe unconfigurable?